Lib/stdlib_deprecations.py

```diff
@@ -0,0 +1,182 @@
+_deprecate('uu', '3.11', remove='3.13'),
+_deprecate('xdrlib', '3.11', remove='3.13'),
+
+# Python 3.12
+_deprecate('datetime.datetime.utcnow', '3.12',
+    replace='datetime.datetime.now(tz=datetime.UTC)'),
+_deprecate('datetime.datetime.utcfromtimestamp', '3.12',
+    replace='datetime.datetime.fromtimestamp(tz=datetime.UTC)'),
+_deprecate('calendar.January', '3.12'),
+_deprecate('calendar.February', '3.12'),
+_deprecate('sys.last_value', '3.12'),
+_deprecate('sys.last_traceback', '3.12'),
+_deprecate('sys.last_exc', '3.12'),
+_deprecate('xml.etree.ElementTree.__bool__', '3.12'),
+
+# Python 3.13
+_deprecate('ctypes.SetPointerType', '3.13', remove='3.15'),
+_deprecate('ctypes.ARRAY', '3.13', remove='3.15'),
+_deprecate('wave.Wave_read.getmark', '3.13', remove='3.15'),
+_deprecate('wave.Wave_read.getmarkers', '3.13', remove='3.15'),
+_deprecate('wave.Wave_read.setmark', '3.13', remove='3.15'),
+
+
+# C API: Python 3.10
+for name in (
+    'PyUnicode_AS_DATA',
+    'PyUnicode_AS_UNICODE',
+    'PyUnicode_AsUnicodeAndSize',
+    'PyUnicode_AsUnicode',
+    'PyUnicode_FromUnicode',
+    'PyUnicode_GET_DATA_SIZE',
+    'PyUnicode_GET_SIZE',
+    'PyUnicode_GetSize',
+    'PyUnicode_IS_COMPACT',
+    'PyUnicode_IS_READY',
+    'PyUnicode_READY',
+    'Py_UNICODE_WSTR_LENGTH',
+    '_PyUnicode_AsUnicode',
+    'PyUnicode_WCHAR_KIND',
+    'PyUnicodeObject',
+    'PyUnicode_InternImmortal',
+):
+    _deprecate_capi(name, '3.10', remove='3.12')
+
+# C API: Python 3.12
+_deprecate_capi('PyDictObject.ma_version_tag', '3.12', remove='3.14'),
+for name, replace in (
+    ('Py_DebugFlag', 'PyConfig.parser_debug'),
+    ('Py_VerboseFlag', 'PyConfig.verbose'),
+    ('Py_QuietFlag', 'PyConfig.quiet'),
+    ('Py_InteractiveFlag', 'PyConfig.interactive'),
+    ('Py_InspectFlag', 'PyConfig.inspect'),
+    ('Py_OptimizeFlag', 'PyConfig.optimization_level'),
+    ('Py_NoSiteFlag', 'PyConfig.site_import'),
+    ('Py_BytesWarningFlag', 'PyConfig.bytes_warning'),
+    ('Py_FrozenFlag', 'PyConfig.pathconfig_warnings'),
+    ('Py_IgnoreEnvironmentFlag', 'PyConfig.use_environment'),
+    ('Py_DontWriteBytecodeFlag', 'PyConfig.write_bytecode'),
+    ('Py_NoUserSiteDirectory', 'PyConfig.user_site_directory'),
+    ('Py_UnbufferedStdioFlag', 'PyConfig.buffered_stdio'),
+    ('Py_HashRandomizationFlag', 'PyConfig.hash_seed'),
+    ('Py_IsolatedFlag', 'PyConfig.isolated'),
+    ('Py_LegacyWindowsFSEncodingFlag', 'PyPreConfig.legacy_windows_fs_encoding'),
+    ('Py_LegacyWindowsStdioFlag', 'PyConfig.legacy_windows_stdio'),
+    ('Py_FileSystemDefaultEncoding', 'PyConfig.filesystem_encoding'),
+    ('Py_FileSystemDefaultEncodeErrors', 'PyConfig.filesystem_errors'),
+    ('Py_UTF8Mode', 'PyPreConfig.utf8_mode'),
+):
+    _deprecate_capi(name, '3.12', replace=replace)
+
+_deprecate_capi('PyErr_Display', '3.12', replace='PyErr_DisplayException()'),
+_deprecate_capi('_PyErr_ChainExceptions', '3.12', replace='_PyErr_ChainExceptions1()'),
+
+
+def is_deprecated(name):
```

Comment:
I think that some linters might what to get `Deprecated` instance back: to get the details, since when, when it is going to be removed. Should we provide an API for that? 🤔

Reply:
> My current API doesn't provide all data. Maybe the API should be get_deprecated() instead: return the dataclass, or None if it's not deprecated.

Yes, I second that :)